### DISCUSSION TITLE
Create helper method for app release

### DISCUSSION THREAD

**Discussion by nafu** [fastlane/fastlane#17246] [Ideas]
# Background

From https://github.com/fastlane/fastlane/issues/17140#issuecomment-693142377 😃 
It would be easier if we have `release` method in `Spaceship::ConnectAPI::App` as we have before in `Spaceship::Tunes::Application` https://github.com/fastlane/fastlane/blob/032ee54518b927e0839a6bc1398124bb08262399/spaceship/lib/spaceship/tunes/application.rb#L386-L396

# Idea

Add `release!` to `Spaceship::ConnectAPI::App` because It's consistent with TunesAPI and familiar to fastlane(spaceship) users.
@joshdholtz What do you think?

## Sample Implementation

```ruby
def release!(platform: nil)
  platform ||= Spaceship::ConnectAPI::Platform::IOS
  version = app.get_pending_release_app_store_version(platform: platform)
  if version.nil?
    raise "Could not find a valid version to release"
  end
  Spaceship::ConnectAPI.post_app_store_version_release_request(app_store_version_id: version.id)
end 
```

### Current

```ruby
require ‘spaceship’
Spaceship::ConnectAPI.login(“your@email.com”, team_name: “Your Team Name”)

app = Spaceship::ConnectAPI::App.find(“com.bundle.id”)

# Without phased release
version = app. get_pending_release_app_store_version(platform: Spaceship::ConnectAPI::Platform::IOS)
Spaceship::ConnectAPI.post_app_store_version_release_request(app_store_version_id: version.id)
```

### With helper method

```ruby
require ‘spaceship’
Spaceship::ConnectAPI.login(“your@email.com”, team_name: “Your Team Name”)

app = Spaceship::ConnectAPI::App.find(“com.bundle.id”)
app.release!
```

# Other consideration

- What about phased release?

**Comment by ainame** [comment]
+1 for having a helper to support this flow.

We've been using this code bit instead of calling `Spaceship::ConnectAPI.post_app_store_version_release_request` directly.

```ruby
app = Spaceship::ConnectAPI::App.find(“com.bundle.id”)
pending_release_version = app.get_pending_release_app_store_version
pending_release_version.create_app_store_version_release_request
``` 

However, the thing is that apparently there's no official support of `Spaceship::ConnectAPI.post_app_store_version_release_request` in App Store Connect API when it comes to using JWT for auth recently introduced in fastlane... Meaning that we would need to use Phased Release APIs even if it's immediate 100% release.

https://github.com/fastlane/fastlane/issues/17283

I haven't even tried JWT yet but it would be this API with `phasedReleaseState: "COMPLETE"` param, I think.

https://developer.apple.com/documentation/appstoreconnectapi/modify_an_app_store_version_phased_release
https://developer.apple.com/documentation/appstoreconnectapi/phasedreleasestate
